numpy/f2py/_backends/_meson.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import errno
+import shutil
+import subprocess
+from pathlib import Path
+
+from ._backend import Backend
+from string import Template
+
+import warnings
+
+
+class MesonTemplate:
+    """Template meson build file generation class."""
+
+    def __init__(
+        self,
+        modulename: str,
+        sources: list[Path],
+        deps: list[str],
+        object_files: list[Path],
+        linker_args: list[str],
+        c_args: list[str],
+    ):
+        self.modulename = modulename
+        self.build_template_path = (
+            Path(__file__).parent.absolute() / "src" / "meson.build.src"
+        )
+        self.sources = sources
+        self.deps = deps
+        self.substitutions = {}
+        self.objects = object_files
+        self.pipeline = [
+            self.initialize_template,
+            self.sources_substitution,
+            self.deps_substitution,
+        ]
+
+    def meson_build_template(self) -> str:
+        if not self.build_template_path.is_file():
+            raise FileNotFoundError(
+                errno.ENOENT,
+                "Meson build template"
+                f" {self.build_template_path.absolute()}"
+                " does not exist.",
+            )
+        return self.build_template_path.read_text()
+
+    def initialize_template(self) -> None:
+        """Initialize with module name."""
+        self.substitutions["modulename"] = self.modulename
+
+    def sources_substitution(self) -> None:
+        indent = " " * 21
+        self.substitutions["source_list"] = f",\n{indent}".join(
+            [f"'{source}'" for source in self.sources]
+        )
+
+    def deps_substitution(self) -> None:
+        indent = " " * 21
+        self.substitutions["dep_list"] = f",\n{indent}".join(
+            [f"dependency('{dep}')" for dep in self.deps]
+        )
+
+    def generate_meson_build(self):
+        for node in self.pipeline:
+            node()
+        template = Template(self.meson_build_template())
+        return template.substitute(self.substitutions)
+
+
+class MesonBackend(Backend):
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            "The MesonBackend is experimental and may change in the future."
```

Comment:
`MesonBackend` -> `Meson backend`

Reply:
More importantly, if this is the default for Python 3.12, it should not emit any warnings. You can just delete it. If there's a bug, it can be fixed normally - that should not affect the user's `f2py -c` invocation.